### PR TITLE
Ft post controller

### PR DESCRIPTION
**Pull request by MSmaili**

### CODE REVIEW

File: src/main/java/com/sda/microblogging/controller/PostController.java [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.bind.annotation.*;
+
+import javax.validation.Valid;
+import javax.validation.constraints.NotNull;
+import java.util.List;
+import java.util.stream.Collectors;
+
+@RestController
+@RequestMapping("/home")
```
Comment (bkrmadtya) [inline src/main/java/com/sda/microblogging/controller/PostController.java]: The base path for post should be post

File: src/main/java/com/sda/microblogging/service/PostService.java [hunk 2]
```diff
@@ -18,19 +19,22 @@
 
     @Autowired
     PostRepository postRepository;
+    @Autowired
+    PostMapper postMapper;
 
     public List<Post> findAll() {
-        return postRepository.findAll();
+        return (List<Post>) postRepository.findAll();
```
Comment (bkrmadtya) [inline src/main/java/com/sda/microblogging/service/PostService.java:27]: How is this line different from the removed line?

**Review by bkrmadtya — DISMISSED** [review]

File: src/main/java/com/sda/microblogging/controller/PostController.java (hunk 1 above)
Comment (Yarulika) [reply src/main/java/com/sda/microblogging/controller/PostController.java]: PostController shall start with
@RequestMapping(path = "/microblogging/v1/post")
Now it even does not appear in swagger..

File: src/test/java/com/sda/microblogging/controller/PostControllerTest.java [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+                .contentType(MediaType.APPLICATION_JSON)
+                .content(asJsonString(null)))
+                .andDo(print());
+
+        resultActions
+                .andExpect(status().isBadRequest());
+    }
+
+    @Test
+    public void find_posts_by_users() throws Exception{
```
Comment (Yarulika) [inline src/test/java/com/sda/microblogging/controller/PostControllerTest.java]: find_posts_by_users or just user?
and what is expected?

File: src/test/java/com/sda/microblogging/controller/PostControllerTest.java [hunk 4]
```diff
@@ -0,0 +1,150 @@
...
+
+
+        List<Post> postDTOS = Arrays.asList(postMapper.convertDtoToPost(new PostSaveDTO()));
+
+        when(postService.findAllPostsAndMyFollowingsPost(1)).thenReturn(postDTOS);
+
+        ResultActions resultActions = mockMvc.perform(get("/home/userId")
+                .param("userId","0")
```
Comment (Yarulika) [inline src/test/java/com/sda/microblogging/controller/PostControllerTest.java]: this param is about absolutely another thing: web request parameter, nothing about test data
like: /books?category=java 
so .param("category", "java")
And test logs explicitly show this:
MockHttpServletRequest:
      HTTP Method = GET
      Request URI = /home/userId 
so it does not get replaced..

**Review by Yarulika — CHANGES_REQUESTED** [review]

**Pull request closed without merging**